cloudbuild-e2e-gce.yaml

```diff
@@ -0,0 +1,37 @@
+# Copyright 2021 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+  # Wait for the image to exist
+  - name: "docker"
+    id: wait-for-image
+    entrypoint: "sh"
+    timeout: 3m
+    env: ["_TEST_SERVER_IMAGE=${_TEST_SERVER_IMAGE}"]
+    args:
+      - e2e-test-server/wait-for-image.sh
+
+  # Run the test
+  - name: $_TEST_RUNNER_IMAGE
+    id: run-tests-gce
+    dir: /
+    env: ["PROJECT_ID=$PROJECT_ID"]
+    args:
+      - gce
+      - --image=$_TEST_SERVER_IMAGE
+
+logsBucket: gs://opentelemetry-ops-e2e-cloud-build-logs
+substitutions:
+  _TEST_RUNNER_IMAGE: gcr.io/opentelemetry-ops-e2e/opentelemetry-operations-e2e-testing:0.5.23
```

Comment:
Is it possible to have some automation for updating this version number, which appears in three YAML files?

Reply:
There are a number of options ive thought of
1. I believe Renovate can be configured to do this, but it would need access to our GCR registry i suppose
2. We could use a docker tag and just push new versions of that tag. I like that the explicit version number makes it easy to re-run tests though
3. I can inject the container version in as a substitution from the "Triggers" which are managed by terraform.
4. We can actually put the entire [Build "yaml" into terraform](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/cloudbuild_trigger#build) and remove all of the duplication. My concerns there are 1) if any repo needs extra build steps and 2) its harder to see what's happening with out of source build config

Reply:
Let's open a bug against the integration test work/component to look into providing this.

Simialrly, the "wait for image" script should be shared in some fashion instead of copy-pasta.